actioncable/lib/action_cable/subscription_adapter/postgresql.rb

```diff
@@ -29,15 +29,20 @@ def shutdown
       end
 
       def with_connection(&block) # :nodoc:
-        ActiveRecord::Base.connection_pool.with_connection do |ar_conn|
-          pg_conn = ar_conn.raw_connection
-
-          unless pg_conn.is_a?(PG::Connection)
-            raise "The Active Record database must be PostgreSQL in order to use the PostgreSQL Action Cable storage adapter"
-          end
+        ar_conn = ActiveRecord::Base.connection_pool.checkout.tap do |conn|
+          # Action Cable is taking ownership over this database connection, and
+          # will perform the necessary cleanup tasks
+          ActiveRecord::Base.connection_pool.remove(conn)
```

Comment:
We only want to do this for the subscription connection, not broadcasts: those should still use AR's `with_connection`.

.. which may be a wrinkle in the viability of this as a backportable solution. :confused: